pkg/apiserver/experiment/experiment.go

```diff
@@ -922,12 +923,108 @@ func (s *Service) getExperimentDetail(c *gin.Context) {
 // @Failure 500 {object} utils.APIError
 // @Router /experiments/{uid} [delete]
 func (s *Service) deleteExperiment(c *gin.Context) {
+	var (
+		chaosKind *v1alpha1.ChaosKind
+		chaosMeta metav1.Object
+		ok        bool
+		exp       *core.Experiment
+	)
+
+	kubeCli, err := clientpool.ExtractTokenAndGetClient(c.Request.Header)
+	if err != nil {
+		_ = c.Error(utils.ErrInvalidRequest.WrapWithNoMessage(err))
+		return
+	}
+
+	uid := c.Param("uid")
+	if exp, err = s.archive.FindByUID(context.Background(), uid); err != nil {
+		if gorm.IsRecordNotFoundError(err) {
+			c.Status(http.StatusInternalServerError)
+			_ = c.Error(utils.ErrInvalidRequest.New("the experiment is not found"))
+		} else {
+			c.Status(http.StatusInternalServerError)
+			_ = c.Error(utils.ErrInternalServer.NewWithNoMessage())
+		}
+		return
+	}
+
+	kind := exp.Kind
+	ns := exp.Namespace
+	name := exp.Name
+	force := c.DefaultQuery("force", "false")
+
+	ctx := context.TODO()
+	chaosKey := types.NamespacedName{Namespace: ns, Name: name}
+
+	if chaosKind, ok = v1alpha1.AllKinds()[kind]; !ok {
+		c.Status(http.StatusBadRequest)
+		_ = c.Error(utils.ErrInvalidRequest.New(kind + " is not supported"))
+		return
+	}
+	if err := kubeCli.Get(ctx, chaosKey, chaosKind.Chaos); err != nil {
+		if apierrors.IsNotFound(err) {
+			c.Status(http.StatusNotFound)
+			_ = c.Error(utils.ErrNotFound.NewWithNoMessage())
+		} else {
+			c.Status(http.StatusInternalServerError)
+			_ = c.Error(utils.ErrInternalServer.WrapWithNoMessage(err))
+		}
+		return
+	}
+
+	if force == "true" {
+		if chaosMeta, ok = chaosKind.Chaos.(metav1.Object); !ok {
+			c.Status(http.StatusInternalServerError)
+			_ = c.Error(utils.ErrInternalServer.WrapWithNoMessage(fmt.Errorf("failed to get chaos meta information")))
+			return
+		}
+
+		annotations := chaosMeta.GetAnnotations()
+		if annotations == nil {
+			annotations = make(map[string]string)
+		}
+		annotations[common.AnnotationCleanFinalizer] = common.AnnotationCleanFinalizerForced
+		chaosMeta.SetAnnotations(annotations)
+		if err := kubeCli.Update(context.Background(), chaosKind.Chaos); err != nil {
```

Comment:
Why don't we need `retryOnConflict` here?

Reply:
@AsterNighT Let me see how to deal with it.

Reply:
I will submit a pr again to solve this problem later, because there is still a forced deletion bug to be dealt with, and it can be dealt with together at that time.